plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Exception.hs

```diff
@@ -144,10 +146,18 @@ mapCauseInMachineException f = bimap (fmap (fmap f)) f
 
 -- | "Prismatically" throw an error and its (optional) cause.
 throwingWithCause
-    :: MonadError (ErrorWithCause e term) m
+    :: forall ex e t term m x
+    . (ex ~ ErrorWithCause e term, MonadError ex m)
     => AReview e t -> t -> Maybe term -> m x
 throwingWithCause l t cause = reviews l (\e -> throwError $ ErrorWithCause e cause) t
 
+-- | "Prismatically" throw an error and its (optional) cause.
+throwingWithCauseEx
```

Comment:
The "Ex" part is very confusing. I assume it means "exception", not our time/space units?

Reply:
Yeah, this stuff is terrible. I just wanted to indicate that it was an exception variant. Maybe I can just get rid of the other one now.

Reply:
yeah sure